### DISCUSSION TITLE
NewLines normalization before or after custom scrubbers?

### DISCUSSION THREAD

**Discussion by JanKrivanek** [VerifyTests/Verify#710] [General]
I'm trying to see if this was intention or not.

Currently the new lines normalization is being done after the custom scrubbers are invoked: https://github.com/VerifyTests/Verify/blob/f5af3cd875ccf0bb55e2338ef9faffd5a6defc75/src/Verify/Serialization/Scrubbers/ApplyScrubbers.cs#L139
It used to be applied before: https://github.com/VerifyTests/Verify/blob/a2540cb684585dffe12031b2a5c860ba0d20f677/src/Verify/Verifier/InnerVerifier_Inner.cs#L68

We made our code less dependent on this behavior (which is good idea anyways) - so I'm just being curious.

**Comment by SimonCropp** [comment]
i think it is reasonable that newlines are normalized after scrubbers. but i cant seem to reproduce the problem

i added some more tests https://github.com/VerifyTests/Verify/commit/312e2370720e54ae88b3a7e994003d3af29eb024

**Comment by JanKrivanek** [comment]
Makes sense.
It is good as it is now - and makes more sense now (applying builtin scrubbers after the custom ones)

It was just an issues in our legacy code that was using custom scrubber that expected the newlines to be already normlized - and hence it got broken by this change. This is fixed on our side already.
